In [1]:
import pandas as pd 
import numpy as np
%matplotlib inline
import numpy as np
import matplotlib.pyplot as plt
plt.style.use('seaborn-white')
from listas import North_America, Cen_Sth_America, Europe, Eurasia, Africa, Middle_East, Asia_Oceania, lst_SRC_C
from variables import dicc_country, ltW, cdW

# Consumo 

df_2 = pd.read_csv('Total_Energy_Consumption.csv', sep=',', header = 1, engine='python', index_col=False)
df_2 = df_2.rename(columns= {'API' : 'Code',
'Unnamed: 1' : 'Source'})


In [2]:
df_2.head(20)

Code  \
0                       NaN   
1     INTL.44-2-WORL-QBTU.A   
2   INTL.4411-2-WORL-QBTU.A   
3   INTL.4413-2-WORL-QBTU.A   
4   INTL.4415-2-WORL-QBTU.A   
5   INTL.4419-2-WORL-QBTU.A   
6   INTL.4417-2-WORL-QBTU.A   
7   INTL.4418-2-WORL-QBTU.A   
8                       NaN   
9      INTL.44-2-AFG-QBTU.A   
10   INTL.4411-2-AFG-QBTU.A   
11   INTL.4413-2-AFG-QBTU.A   
12   INTL.4415-2-AFG-QBTU.A   
13   INTL.4419-2-AFG-QBTU.A   
14                     none   
15   INTL.4418-2-AFG-QBTU.A   
16                      NaN   
17     INTL.44-2-ALB-QBTU.A   
18   INTL.4411-2-ALB-QBTU.A   
19   INTL.4413-2-ALB-QBTU.A   

                                               Source                   1980  \
0                                               World                    NaN   
1                              Consumption (quad Btu)     292.89978962044677   
2                                     Coal (quad Btu)      78.65613402957176   
3                              Natural gas (quad Btu)           53.865223303   
4              Petroleum and other liquids (quad Btu)     132.06401937163076   
5           Nuclear, renewables, and other (quad Btu)     28.278044616244237   
6                                  Nuclear (quad Btu)      7.575700462108056   
7                     Renewables and other (quad Btu)      20.70234415413618   
8                                         Afghanistan                    NaN   
9                              Consumption (quad Btu)    0.02658321667756184   
10                                    Coal (quad Btu)  0.0024792482148388983   
11                             Natural gas (quad Btu)               0.002094   
12             Petroleum and other liquids (quad Btu)    0.01462409812353294   
13          Nuclear, renewables, and other (quad Btu)       0.00738587033919   
14                                 Nuclear (quad Btu)                    NaN   
15                    Renewables and other (quad Btu)       0.00738587033919   
16                                            Albania                    NaN   
17                             Consumption (quad Btu)    0.16298182182124926   
18                                    Coal (quad Btu)   0.024317314523581157   
19                             Natural gas (quad Btu)                0.01047   

                     1981                   1982                   1983  \
0                     NaN                    NaN                    NaN   
1      289.40102169937586      289.6911022148106     293.73162574512844   
2        78.9635811572502      80.42233145882604      82.63117397819225   
3            53.895423471           54.055255448           55.149772666   
4      126.81113213676979     123.92428288938333     122.21533499685697   
5        29.6404771213559     31.229996382601204      33.66568072507923   
6       8.527153469041346       9.50768642751678     10.718344095623602   
7      21.113323652314552     21.722309955084427     22.947336629455624   
8                     NaN                    NaN                    NaN   
9    0.029949457623843767   0.031897814673808554    0.03902180313163165   
10  0.0026042523630414455  0.0030209327940489256  0.0030209327940489256   
11               0.003141               0.005235               0.005235   
12    0.01666759037178232    0.01625090060634963   0.022917936760382725   
13       0.00753661488902       0.00739098127341        0.0078479335772   
14                    NaN                    NaN                    NaN   
15       0.00753661488902       0.00739098127341        0.0078479335772   
16                    NaN                    NaN                    NaN   
17    0.16204449273682853    0.16284611700217974    0.16933585149179003   
18   0.025770486274170428   0.029070074678547934    0.03146270929689194   
19               0.011517                0.01047               0.011517   

                     1984                  1985                   1986  \
0                     NaN               

In [3]:
len(df_2['Source'])

1848

In [4]:
len(df_2['Source']) / 8

231.0

In [5]:
df_2['Source'] = ['Country', 'Consumption', 'Coal', 'Natural gas', 'Petroleum and other liquids', 'Nuclear, renewables, and other', 'Nuclear', 'Renewables'] * 231

In [6]:
df_2.head(20)

Code                          Source  \
0                       NaN                         Country   
1     INTL.44-2-WORL-QBTU.A                     Consumption   
2   INTL.4411-2-WORL-QBTU.A                            Coal   
3   INTL.4413-2-WORL-QBTU.A                     Natural gas   
4   INTL.4415-2-WORL-QBTU.A     Petroleum and other liquids   
5   INTL.4419-2-WORL-QBTU.A  Nuclear, renewables, and other   
6   INTL.4417-2-WORL-QBTU.A                         Nuclear   
7   INTL.4418-2-WORL-QBTU.A                      Renewables   
8                       NaN                         Country   
9      INTL.44-2-AFG-QBTU.A                     Consumption   
10   INTL.4411-2-AFG-QBTU.A                            Coal   
11   INTL.4413-2-AFG-QBTU.A                     Natural gas   
12   INTL.4415-2-AFG-QBTU.A     Petroleum and other liquids   
13   INTL.4419-2-AFG-QBTU.A  Nuclear, renewables, and other   
14                     none                         Nuclear   
15   INTL.4418-2-AFG-QBTU.A                      Renewables   
16                      NaN                         Country   
17     INTL.44-2-ALB-QBTU.A                     Consumption   
18   INTL.4411-2-ALB-QBTU.A                            Coal   
19   INTL.4413-2-ALB-QBTU.A                     Natural gas   

                     1980                   1981                   1982  \
0                     NaN                    NaN                    NaN   
1      292.89978962044677     289.40102169937586      289.6911022148106   
2       78.65613402957176       78.9635811572502      80.42233145882604   
3            53.865223303           53.895423471           54.055255448   
4      132.06401937163076     126.81113213676979     123.92428288938333   
5      28.278044616244237       29.6404771213559     31.229996382601204   
6       7.575700462108056      8.527153469041346       9.50768642751678   
7       20.70234415413618     21.113323652314552     21.722309955084427   
8                     NaN                    NaN                    NaN   
9     0.02658321667756184   0.029949457623843767   0.031897814673808554   
10  0.0024792482148388983  0.0026042523630414455  0.0030209327940489256   
11               0.002094               0.003141               0.005235   
12    0.01462409812353294    0.01666759037178232    0.01625090060634963   
13       0.00738587033919       0.00753661488902       0.00739098127341   
14                    NaN                    NaN                    NaN   
15       0.00738587033919       0.00753661488902       0.00739098127341   
16                    NaN                    NaN                    NaN   
17    0.16298182182124926    0.16204449273682853    0.16284611700217974   
18   0.024317314523581157   0.025770486274170428   0.029070074678547934   
19                0.01047               0.011517                0.01047   

                     1983                   1984                  1985  \
0                     NaN                    NaN                   NaN   
1      293.73162574512844      307.2343654436228     315.4271480750418   
2       82.63117397819225      86.35602132998221     89.40927570308125   
3            55.149772666           60.842896415          63.371661785   
4      122.21533499685697     123.41709237538814    123.47159748556288   
5       33.66568072507923      36.53782339725242    39.100817440397684   
6      10.718344095623602     12.994607878980153    15.298615144911766   
7      22.947336629455624     23.543215518272262     23.80220229548592   
8                     NaN                    NaN                   NaN   
9     0.03902180313163165    0.03920224040186807   0.03827496129519434   
10  0.0030209327940489256  0.0030834347736486832  0.003145936942251473   
11               0.005235               0.005235                     0   
12   0.022917936760382725    0.02298072562821939  0.027084834352942865   
13        0.0078479335772             0.00790308            0.00804419   
14           

In [7]:
df_2.tail(20)

Code                          Source  \
1828  INTL.4415-2-YEM-QBTU.A     Petroleum and other liquids   
1829  INTL.4419-2-YEM-QBTU.A  Nuclear, renewables, and other   
1830                    none                         Nuclear   
1831  INTL.4418-2-YEM-QBTU.A                      Renewables   
1832                     NaN                         Country   
1833    INTL.44-2-ZMB-QBTU.A                     Consumption   
1834  INTL.4411-2-ZMB-QBTU.A                            Coal   
1835  INTL.4413-2-ZMB-QBTU.A                     Natural gas   
1836  INTL.4415-2-ZMB-QBTU.A     Petroleum and other liquids   
1837  INTL.4419-2-ZMB-QBTU.A  Nuclear, renewables, and other   
1838                    none                         Nuclear   
1839  INTL.4418-2-ZMB-QBTU.A                      Renewables   
1840                     NaN                         Country   
1841    INTL.44-2-ZWE-QBTU.A                     Consumption   
1842  INTL.4411-2-ZWE-QBTU.A                            Coal   
1843  INTL.4413-2-ZWE-QBTU.A                     Natural gas   
1844  INTL.4415-2-ZWE-QBTU.A     Petroleum and other liquids   
1845  INTL.4419-2-ZWE-QBTU.A  Nuclear, renewables, and other   
1846                    none                         Nuclear   
1847  INTL.4418-2-ZWE-QBTU.A                      Renewables   

                      1980                  1981                  1982  \
1828   0.09549226626503785   0.09734761084333841    0.1142776301185542   
1829                     0                     0                     0   
1830                   NaN                   NaN                   NaN   
1831                     0                     0                     0   
1832                   NaN                   NaN                   NaN   
1833   0.13311664167592396    0.1324782683524794     0.135388943093979   
1834  0.015136894032273961  0.012094153254484357  0.013189697449913943   
1835                     0                     0                     0   
1836  0.033729306000750005  0.029432505746875046  0.029432505746875046   
1837       0.0842504416429      0.09095160935112      0.09276673989719   
1838                   NaN                   NaN                   NaN   
1839       0.0842504416429      0.09095160935112      0.09276673989719   
1840                   NaN                   NaN                   NaN   
1841    0.1414723741419828    0.1423668569379908   0.13490705532101982   
1842   0.06621009071857202   0.06104049502561372   0.06327173087484642   
1843                     0                     0                     0   
1844  0.026817329215339573  0.030858528262911643  0.024686822608620606   
1845  0.048444954208071205   0.05046783364946544    0.0469485018375528   
1846                   NaN                   NaN                   NaN   
1847  0.048444954208071205   0.05046783364946544    0.0469485018375528   

                      1983                  1984                  1985  \
1828   0.10581262048162848   0.13156712240979637    0.1460214162647722   
1829                     0                     0                     0   
1830                   NaN                   NaN                   NaN   
1831                     0                     0                     0   
1832                   NaN                   NaN                   NaN   
1833   0.13482301200356028   0.13046882774657492   0.12923196814160648   
1834  0.011776885530685234  0.011356318624572684  0.011589222073092067   
1835                     0                     0                     0   
1836  0.029432505746875046  0.027405061122002243  0.025227862068514406   
1837        0.093613620726           0.091707448           0.092414884   
1838                   NaN                   NaN                   NaN   
1839        0.093613620726           0.091707448           0.092414884   
1840                   NaN                   NaN                   NaN   
1841   0.13931682987471733    0.1393448630494655    0.1464299031845133   
1842    0.0620

In [8]:

'''Limpiamos nuestras columnas y nos quedamos únicamente con los países de nuestro DataSet y sus acrónimos'''

#def clean_data():
# Una vez sacamos una lista de totdos los países eliminamos la filas con NAn
df_2 = df_2.dropna(axis=0, thresh=39)

#Lista de Países
# lista_Paises

lista_1 = []
lista_2 = []

for n in df_2['Code']:
    string = n.split('-')
    lista_1.append(string)
# print(lista_1)
for m in lista_1:
    elemento = m[2]
    lista_2.append(elemento)
# print(lista_2)
#Modificamos la columna 'Code'
df_2['Code'] = lista_2


In [9]:

'''Modificamos la columna 'Source'; Recordar que las unidades son (quad Btu)'''

# arr = df_2['Source'].unique()
# lst_SRC_old = arr.tolist()
# src = zip(lst_SRC_new, lst_SRC_old)
# source = list(src)

# for list in source:
#     df_2.replace(list[1], 
#            list[0], 
#            inplace=True)

#to numeric
# dfs.loc[:,'1980':'2019'] = dfs.loc[:,'1980':'2019'].apply(pd.to_numeric())
for col in df_2.loc[:,'1980':'2019']:
    df_2[col] = pd.to_numeric(df_2[col], errors ='coerce')

'''df_1 es la versión extendidad de dfP'''

df_2.replace(0.0, np.nan, inplace=True)
df_2 = df_2.dropna(thresh=4).reset_index(drop=True)
df_2 = df_2.fillna(0)

'''Creamos dfC que es el DataSet con el que vamos a trabajar (desde 1997 hasta 2019)'''

dfC = df_2.drop(df_2.loc[:, '1980':'1996'], axis=1)
dfC.replace(0.0, np.nan, inplace=True)
dfC = dfC.dropna(thresh=4).reset_index(drop=True)
dfC = dfC.fillna(0)

## df_2 Consumo

In [10]:
# dfC.head(10)
#dfC.tail(10)
# dfC.shape
dfC.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1146 entries, 0 to 1145
Data columns (total 25 columns):
 #   Column  Non-Null Count  Dtype  
---  ------  --------------  -----  
 0   Code    1146 non-null   object 
 1   Source  1146 non-null   object 
 2   1997    1146 non-null   float64
 3   1998    1146 non-null   float64
 4   1999    1146 non-null   float64
 5   2000    1146 non-null   float64
 6   2001    1146 non-null   float64
 7   2002    1146 non-null   float64
 8   2003    1146 non-null   float64
 9   2004    1146 non-null   float64
 10  2005    1146 non-null   float64
 11  2006    1146 non-null   float64
 12  2007    1146 non-null   float64
 13  2008    1146 non-null   float64
 14  2009    1146 non-null   float64
 15  2010    1146 non-null   float64
 16  2011    1146 non-null   float64
 17  2012    1146 non-null   float64
 18  2013    1146 non-null   float64
 19  2014    1146 non-null   float64
 20  2015    1146 non-null   float64
 21  2016    1146 non-null   float64
 22  

In [11]:
dfC.describe()

1997         1998         1999         2000         2001  \
count  1146.000000  1146.000000  1146.000000  1146.000000  1146.000000   
mean      1.422183     1.430240     1.457980     1.504114     1.514898   
std      13.268334    13.345018    13.598362    14.031566    14.112850   
min      -0.070028    -0.072393    -0.069423    -0.097200    -0.099622   
25%       0.001816     0.001919     0.002191     0.002377     0.002726   
50%       0.029838     0.031732     0.034655     0.035043     0.035207   
75%       0.266214     0.278044     0.263862     0.283462     0.285168   
max     380.356940   382.488941   389.689924   402.378722   405.220362   

              2002         2003         2004         2005         2006  ...  \
count  1146.000000  1146.000000  1146.000000  1146.000000  1146.000000  ...   
mean      1.546327     1.597069     1.679272     1.733534     1.788192  ...   
std      14.412141    14.918307    15.707866    16.221328    16.738744  ...   
min      -0.108294    -0.155209    -0.137971    -0.120652    -0.163438  ...   
25%       0.002869     0.003081     0.003226     0.003401     0.003615  ...   
50%       0.037273     0.037495     0.039878     0.042696     0.045174  ...   
75%       0.300156     0.314416     0.329359     0.333756     0.333542  ...   
max     413.970911   428.501632   450.689142   465.293106   480.024628  ...   

              2010         2011         2012         2013         2014  \
count  1146.000000  1146.000000  1146.000000  1146.000000  1146.000000   
mean      1.966152     2.015527     2.054602     2.094244     2.116646   
std      18.421125    18.947476    19.365283    19.712948    19.886830   
min      -0.071121    -0.066841    -0.023114    -0.031287    -0.010924   
25%       0.004712     0.005129     0.005614     0.005869     0.005801   
50%       0.053899     0.052157     0.053428     0.058235     0.055731   
75%       0.315410     0.339376     0.347172     0.347472     0.353918   
max     526.730468   540.495127   551.600383   561.092151   566.129942   

              2015         2016         2017         2018         2019  
count  1146.000000  1146.000000  1146.000000  1146.000000  1146.000000  
mean      2.122756     2.145388     2.192994     2.247041     2.263406  
std      19.905007    20.056177    20.471183    20.951856    21.082750  
min      -0.010894    -0.010891    -0.010891    -0.010891    -0.010916  
25%       0.005653     0.006441     0.006875     0.007031     0.008011  
50%       0.059169     0.066347     0.069060     0.069980     0.070253  
75%       0.346879     0.360374     0.368329     0.377442     0.377281  
max     567.328822   572.159231   584.201816   597.736582   601.040490  

[8 rows x 23 columns]

In [12]:
dfC.iloc[120:130]

Code                          Source      1997      1998      1999  \
120  BTN                      Renewables  0.014135  0.013592  0.013569   
121  BOL                     Consumption  0.135993  0.126144  0.141700   
122  BOL                     Natural gas  0.049452  0.032107  0.033214   
123  BOL     Petroleum and other liquids  0.070125  0.078231  0.089769   
124  BOL  Nuclear, renewables, and other  0.016417  0.015806  0.018717   
125  BOL                      Renewables  0.016417  0.015806  0.018717   
126  BIH                     Consumption  0.166312  0.196414  0.199242   
127  BIH                            Coal  0.083225  0.107370  0.103356   
128  BIH                     Natural gas  0.004905  0.007006  0.007006   
129  BIH     Petroleum and other liquids  0.038958  0.043804  0.044844   

         2000      2001      2002      2003      2004  ...      2010  \
120  0.013578  0.015127  0.016913  0.019099  0.017858  ...  0.052196   
121  0.148727  0.127276  0.138827  0.186631  0.193555  ...  0.230194   
122  0.046499  0.030261  0.037642  0.082665  0.078975  ...  0.100379   
123  0.081699  0.074641  0.078416  0.083472  0.092698  ...  0.107961   
124  0.020529  0.022374  0.022769  0.020494  0.021882  ...  0.021854   
125  0.020529  0.022374  0.022769  0.020494  0.021882  ...  0.021854   
126  0.220676  0.219539  0.220654  0.192517  0.198827  ...  0.261047   
127  0.113506  0.111827  0.119534  0.101749  0.090320  ...  0.130575   
128  0.010510  0.010510  0.005605  0.005605  0.010510  ...  0.008252   
129  0.048523  0.047920  0.046263  0.043454  0.050856  ...  0.071584   

         2011      2012      2013      2014      2015      2016      2017  \
120  0.049921  0.047688  0.053281  0.050984  0.052967  0.053388  0.051357   
121  0.238262  0.282418  0.286214  0.288699  0.303434  0.314244  0.309750   
122  0.088570  0.117724  0.124367  0.118020  0.099641  0.117650  0.117503   
123  0.126257  0.141731  0.137002  0.148531  0.179031  0.178868  0.170035   
124  0.023435  0.022962  0.024845  0.022149  0.024761  0.017725  0.022213   
125  0.023435  0.022962  0.024845  0.022149  0.024761  0.017725  0.022213   
126  0.252929  0.246045  0.249843  0.241865  0.242228  0.247680  0.245790   
127  0.144656  0.136840  0.130476  0.135799  0.134452  0.136395  0.145681   
128  0.009329  0.008611  0.008611  0.006064  0.007176  0.007499  0.008145   
129  0.073180  0.066107  0.061869  0.063017  0.064333  0.074407  0.072630   

         2018      2019  
120  0.047559  0.064091  
121  0.319736  0.311933  
122  0.118162  0.107717  
123  0.174808  0.171089  
124  0.026766  0.033127  
125  0.026766  0.033127  
126  0.249930  0.237304  
127  0.141078  0.126302  
128  0.008131  0.007831  
129  0.071416  0.070664  

[10 rows x 25 columns]

## df_3 Población

In [13]:
# Nivel poblacional Mundial -> Population (Mperson)

df_3 = pd.read_csv('Population.csv', sep=',', header = 1, engine='python', index_col=False)
df_3 = df_3.rename(columns= {'API' : 'Code',
'Unnamed: 1' : 'Country'})

# Una vez sacamos una lista de totdos los países eliminamos la filas con NAn
df_3 = df_3.dropna(axis=0, thresh=39)
# df_2.head(10)

#Lista de Países
# lista_Paises

lista_1 = []
lista_2 = []

for n in df_3['Code']:
    string = n.split('-')
    lista_1.append(string)
# print(lista_1)
for m in lista_1:
    elemento = m[2]
    lista_2.append(elemento)
# print(lista_2)

df_3['Code'] = lista_2

#to numeric
# dfs.loc[:,'1980':'2019'] = dfs.loc[:,'1980':'2019'].apply(pd.to_numeric())
for col in df_3.loc[:,'1980':'2019']:
    df_3[col] = pd.to_numeric(df_3[col], errors ='coerce')

df_3 = df_3.fillna(method="bfill", axis=1)
df_3 = df_3.fillna(method="ffill", axis=1) 
dfNP = df_3.drop(df_3.loc[:, '1980':'1996'], axis=1)

for col in dfNP.loc[:,'1997':'2020']:
    dfNP[col] = pd.to_numeric(dfNP[col], errors ='coerce')


In [14]:
#dfNP.head(10)
#dfNP.tail(10)
# dfNP.shape
# dfNP.info()

In [15]:
dfNP.describe()

1997          1998          1999          2000          2001  \
count  1.980000e+02  1.980000e+02  1.980000e+02  1.980000e+02  1.980000e+02   
mean   5.946062e+04  6.025509e+04  6.105126e+04  6.184370e+04  6.263478e+04   
std    4.339915e+05  4.397203e+05  4.454429e+05  4.511442e+05  4.568374e+05   
min    1.966000e+01  1.982000e+01  2.002600e+01  2.031300e+01  2.067500e+01   
25%    1.293148e+03  1.291085e+03  1.308152e+03  1.336463e+03  1.365930e+03   
50%    5.656420e+03  5.748520e+03  5.834885e+03  5.916940e+03  6.017655e+03   
75%    1.846388e+04  1.861399e+04  1.883014e+04  1.922357e+04  1.964249e+04   
max    5.906491e+06  5.985041e+06  6.063582e+06  6.141881e+06  6.220161e+06   

               2002          2003          2004          2005          2006  \
count  1.980000e+02  1.980000e+02  1.980000e+02  1.980000e+02  1.980000e+02   
mean   6.342745e+04  6.422254e+04  6.502447e+04  6.583441e+04  6.665703e+04   
std    4.625350e+05  4.682453e+05  4.739995e+05  4.798071e+05  4.856999e+05   
min    2.112800e+01  2.167400e+01  2.232900e+01  2.310600e+01  2.402200e+01   
25%    1.382457e+03  1.387170e+03  1.393267e+03  1.428847e+03  1.470465e+03   
50%    6.133965e+03  6.250875e+03  6.375380e+03  6.512185e+03  6.663765e+03   
75%    2.006515e+04  2.049876e+04  2.094442e+04  2.110147e+04  2.115349e+04   
max    6.298597e+06  6.377289e+06  6.456653e+06  6.536810e+06  6.618224e+06   

       ...          2011          2012          2013          2014  \
count  ...  1.980000e+02  1.980000e+02  1.980000e+02  1.980000e+02   
mean   ...  7.088578e+04  7.173855e+04  7.260306e+04  7.346641e+04   
std    ...  5.162449e+05  5.223321e+05  5.284840e+05  5.346239e+05   
min    ...  2.832600e+01  2.865400e+01  2.885000e+01  2.898500e+01   
25%    ...  1.805285e+03  1.831587e+03  1.854517e+03  1.909375e+03   
50%    ...  7.407627e+03  7.457753e+03  7.497542e+03  7.536363e+03   
75%    ...  2.409253e+04  2.471875e+04  2.509270e+04  2.563208e+04   
max    ...  7.041713e+06  7.126262e+06  7.211822e+06  7.297270e+06   

               2015          2016          2017          2018          2019  \
count  1.980000e+02  1.980000e+02  1.980000e+02  1.980000e+02  1.980000e+02   
mean   7.429451e+04  7.515143e+04  7.600304e+04  7.685059e+04  7.768282e+04   
std    5.405208e+05  5.466091e+05  5.526581e+05  5.586734e+05  5.645760e+05   
min    2.914800e+01  2.935500e+01  2.956700e+01  2.979030e+01  3.001370e+01   
25%    1.953008e+03  1.964560e+03  1.968075e+03  1.961105e+03  1.962110e+03   
50%    7.650830e+03  7.809465e+03  7.971165e+03  8.136490e+03  8.301815e+03   
75%    2.616938e+04  2.670308e+04  2.711715e+04  2.766990e+04  2.813294e+04   
max    7.379227e+06  7.464043e+06  7.548344e+06  7.632247e+06  7.714631e+06   

               2020  
count  1.980000e+02  
mean   7.848397e+04  
std    5.702477e+05  
min    3.023700e+01  
25%    1.996845e+03  
50%    8.467140e+03  
75%    2.824971e+04  
max    7.793795e+06  

[8 rows x 24 columns]

## df_4 Consumo per capita

In [16]:
# Energy Intensity; Population (MMBtu/person)

'''Información incompleta'''
# df_4 = df_2 = pd.read_csv('KWh_persona.csv', sep=',', header = 2, engine='python', index_col=False)
# df_4.head()
'''Otro tipo de formato'''
# df_EI = pd.read_csv('Energy_Intensity.csv', sep=',', header = 1, engine='python', index_col=False)
# df_EI.T.head(10)

dfEI = pd.read_csv('Energy_Intensity_2.csv', sep=',', header = 1, engine='python', index_col=False)
dfEI = dfEI.rename(columns= {'API' : 'Code',
'Unnamed: 1' : 'Country'})

'''Limpiamos nuestras columnas y nos quedamos únicamente con los países de nuestro DataSet y sus acrónimos'''

#def clean_data():
# Una vez sacamos una lista de totdos los países eliminamos la filas con NAn
dfEI = dfEI.dropna(axis=0, thresh=41)

#Lista de Países
# lista_Paises

lista_1 = []
lista_2 = []

for n in dfEI['Code']:
    string = n.split('-')
    lista_1.append(string)
# print(lista_1)
for m in lista_1:
    elemento = m[2]
    lista_2.append(elemento)
# print(lista_2)
#Modificamos la columna 'Code'
dfEI['Code'] = lista_2

dfEI = dfEI.drop(dfEI.index[1::2])
dfEI = dfEI.drop(['Country'], axis = 1).reset_index(drop=True)

#to numeric
# dfs.loc[:,'1980':'2019'] = dfs.loc[:,'1980':'2019'].apply(pd.to_numeric())
for col in dfEI.loc[:,'1980':'2019']:
    dfEI[col] = pd.to_numeric(dfEI[col], errors ='coerce')

'''df_4 es la versión extendidad de dfEI'''
df_4 = dfEI.copy()
df_4.replace(0.0, np.nan, inplace=True)
df_4 = df_4.dropna(thresh=4).reset_index(drop=True)
df_4 = df_4.fillna(0)

'''Creamos dfEI que es el DataSet con el que vamos a trabajar (desde 1997 hasta 2019)'''

dfEI = dfEI.drop(dfEI.loc[:, '1980':'1996'], axis=1)
dfEI.replace(0.0, np.nan, inplace=True)
dfEI = dfEI.dropna(thresh=4).reset_index(drop=True)
dfEI = dfEI.fillna(0)

In [17]:
#dfEI.head(10)
#dfEI.tail(10)
dfEI.shape
#dfEI.info()

(198, 24)

In [18]:
dfEI.describe()

1997         1998         1999         2000        2001  \
count   198.000000   198.000000   198.000000   198.000000  198.000000   
mean     79.783010    79.646623    79.388300    83.375103   83.221654   
std     127.024047   125.131187   123.363836   128.029261  118.483512   
min       0.000000     0.000000     0.000000     0.000000    0.000000   
25%       6.749172     6.389241     7.009977     7.431312    7.778363   
50%      38.366378    37.685746    36.061853    35.806542   36.100066   
75%     107.105507   104.506885   101.349627   118.480111  121.005792   
max    1139.320598  1127.238553  1082.594019  1115.454340  817.633877   

             2002        2003        2004         2005        2006  ...  \
count  198.000000  198.000000  198.000000   198.000000  198.000000  ...   
mean    84.446598   85.030609   86.867677    88.927116   90.631946  ...   
std    121.214875  120.657268  123.783793   127.970631  125.012388  ...   
min      0.000000    0.000000    0.000000     0.000000    0.000000  ...   
25%      7.447488    7.926378    8.840988     9.535746   10.786571  ...   
50%     38.146424   38.558599   39.311311    40.290972   43.852872  ...   
75%    130.734264  132.430148  129.617715   125.163212  125.623586  ...   
max    939.390068  928.076843  960.557942  1027.044351  946.090378  ...   

             2010        2011        2012        2013        2014        2015  \
count  198.000000  198.000000  198.000000  198.000000  198.000000  198.000000   
mean    88.273683   86.552949   88.150011   87.185866   86.460385   86.636801   
std    110.700924  110.573918  116.572296  116.410329  114.317087  113.314553   
min      0.000000    0.000000    0.744630    0.761845    0.713444    0.698022   
25%     12.212746   12.078931   12.564280   13.781629   13.671984   13.406518   
50%     47.615211   47.368746   51.681729   51.239893   50.711326   51.067970   
75%    121.179874  110.941598  115.159886  111.041720  110.132339  108.313166   
max    675.920290  674.859694  871.199188  884.538071  851.038558  776.435213   

             2016        2017        2018        2019  
count  198.000000  198.000000  198.000000  198.000000  
mean    87.025398   87.950824   87.929409   87.740986  
std    111.800567  113.254565  111.394173  111.688365  
min      0.739355    0.845739    0.825315    0.801967  
25%     11.861526   11.870872   12.505346   12.848532  
50%     52.954609   52.659685   53.244764   53.972640  
75%    107.970733  108.271653  108.426248  108.164481  
max    724.127969  770.484537  705.693884  723.581880  

[8 rows x 23 columns]

## Ejemplo! df_4 Consumo per capita
1 Quad = 293 071,070 172 22 Gigavatio-hora [GWh] - Calculador de unidades de medida, con el cual podemos convertir entre otros Quad en Gigavatio-hora.

In [19]:
# '''Prueba_1'''
# pers = float(df_3[df_3['Code'] == 'LKA']['1980'])
# pers10_3 = pers * 1000
# print(pers10_3)
# num = float(df_2[(df_2['Code'] == 'LKA') & (df_2['Source'] == 'Consumption')]['1980'])
# print(num)
# gwh = num * 293071.07017222
# print(gwh)
# mwh = gwh / pers10_3 * 1000
# print(mwh)
# list = [str(x) for x in range(1980, 2020, 1)]
# # print(list)

# '''Primera Opción'''
# # def consumo_per_capita():
# list_cc = []
# #for i in range(1980, 2020, 1):

# for i in list:
#     # 1.Pillamos el num de personas en Afg en cada año | 2.Pasamos de millones de personas a personas
#     pers = float(df_3[df_3['Code'] == 'LKA'][i]) * 1000
#     # 1.Pillamos el consumo en Btu en Afg para cada año | 2.Pasamos de Btu a Gwh
#     num = float(df_2[(df_2['Code'] == 'LKA') & (df_2['Source'] == 'Consumption')][i])
#     gwh = num * 293071.07017222
#     # 1. Finalmente nos quedamos con Kwh/persona
#     mwh_pers = round(((gwh / pers) * 1000), 3)
#     np.nan_to_num(mwh_pers, copy=False)
#     list_cc.append(mwh_pers)

# list_cc 

# '''Segunda Opción'''
# list = [str(x) for x in range(1980, 2020, 1)]
# df_cc = pd.DataFrame({'year' : list})

# for pais in cdW:
#     list_aux = []
#     for i in list:
#         # 1.Pillamos el num de personas en Afg en cada año | 2.Pasamos de millones de personas a personas
#         pers = df_3[df_3['Code'] == pais][i].astype(float) * 1000
#         # 1.Pillamos el consumo en Btu en Afg para cada año | 2.Pasamos de Btu a Gwh
#         num = df_2[(df_2['Code'] == pais) & (df_2['Source'] == 'Consumption')][i].astype(float)
#         gwh = num * 293071.07017222
#         # 1. Finalmente nos quedamos con Kwh/persona
#         mwh_pers = round(((gwh / pers) * 1000), 3)
#         mwh_pers =str(mwh_pers)
#         # np.nan_to_num(mwh_pers, copy=False)

#         list_aux.append(mwh_pers)
#     df_cc[pais] = list_aux